File: src/service/shared/WebSocketMessages/Messages/Librarians/WebSocketLibrarianList.cs
namespace WebSocketMessages.Messages.Librarians
{

    public class WebSocketLibrainDocRef
    {
        public string? DocumentUri;
        public string? Text ;
        public string? Question;
        public string? Score;
    }

    public class WebSocketLibrarianList : WebSocketBaseMessage
    {
        public string? Question;
        public string? RoomName;
        public string? AgentName;
        public List<WebSocketLibrainDocRef> References = [];
      
    }
}

File: src/client/shared/src/models/WebSocketLibrarianList.ts
import { WebSocketBaseMessage } from './WebSocketBaseMessage';


/** Profile of an agent in a chat room */
export interface WebSocketLibrainDocRef {
    /** Agent's name */
    DocumentUri: string;
    Text: string;
    Question: string;
    Score: string;
  }
  

export interface WebSocketLibrarianList extends WebSocketBaseMessage {
    Question: string; // Base64 encoded audio data
    RoomName: string;
    AgentName: string;
    References: WebSocketLibrainDocRef[];
}
  